broker/broker_utils/broker_utils/beam_transforms.py

```diff
@@ -112,11 +112,23 @@ def process_ztf(self, alertDict):
 class FormatDictForPubSub(DoFn):
     def process(self, alertDict):
         """Converts alert packet dictionaries to format suitable for WriteToPubSub().
-        Currently returns a bytes object (includes msg data only).
-        Can be updated to return a :class:`~PubsubMessage` object.
-        In that case, change WriteToPubSub() kwarg 'with_attributes' to `True`.
-        See https://beam.apache.org/releases/pydoc/2.26.0/apache_beam.io.external.gcp.pubsub.html?highlight=writetopubsub#apache_beam.io.external.gcp.pubsub.WriteToPubSub
+
+        Returns a `PubsubMessage` object with custom attributes attached.
         """
+        from apache_beam.io.gcp.pubsub import PubsubMessage
         import json
+
+        # collect message attributes
```

Comment:
Can you add some context about why sometimes we want the value pointed to by `alertDict['alert']` but if that key doesn't exist we want the full `alertDict`.  Reading this out of context, that sounds like an odd way for the alert data to be organized.